sim/midas/src/main/cc/bridges/termination.cc

```diff
@@ -0,0 +1,67 @@
+#ifdef TERMINATIONBRIDGEMODULE_struct_guard
+
+#include "termination.h"
+#include <iostream>
+
+termination_t::termination_t(
+  simif_t* sim, 
+  std::vector<std::string> &args,
+  TERMINATIONBRIDGEMODULE_struct * mmio_addrs,
+  unsigned int num_messages,
+  unsigned int* isErr,
+  const char* const* msgs): 
+  bridge_driver_t(sim), 
+  mmio_addrs(mmio_addrs), 
+  isErr(isErr),
+  msgs(msgs)
+{
+  //tick-rate to decide sampling rate of MMIOs per number of ticks
+  this->mmio_addrs = mmio_addrs;
+  std::string tick_rate_arg = std::string("+tick-rate=");
+  for (auto &arg: args) {
+    if (arg.find(tick_rate_arg) == 0) {
+      char *str = const_cast<char*>(arg.c_str()) + tick_rate_arg.length();
+      uint64_t tick_period = atol(str);
+      this->tick_rate = tick_period;
+    }
+  }
+}
+
+termination_t::~termination_t() {
+  free(this->mmio_addrs);
+}
+
+void termination_t::tick() {  //reads the MMIOs every tick
+  if(tick_counter == tick_rate) {
+    if (read(this->mmio_addrs->out_status)) {
+      int msg_id = read(this->mmio_addrs->out_errCode);
+    	fail = msg_id;
```

Comment:
This is incorrect. It needs to be:

```suggestion
    	this->fail = isErr[msg_id] != 0; 
```

Your test should catch this.

Reply:
Yup. Redundant.